utils/ansible/ftest/templates/daos-launch.sh.j2

```diff
@@ -0,0 +1,235 @@
+CLIENTS_LIST="{{ groups["daos_clients"] | join(',') }}"
+{% endif %}
+MODULE_OPENMPI_PATH="{{daos_openmpi_module_path }}"
+
+PYTHON_EXE=python3
+CLUSH_EXE=clush
+MODULE_EXE=module
+
+ANSI_COLOR_BLACK=30
+ANSI_COLOR_RED=31
+ANSI_COLOR_GREEN=32
+ANSI_COLOR_YELLOW=33
+ANSI_COLOR_BLUE=34
+ANSI_COLOR_MAGENTA=35
+ANSI_COLOR_CYAN=36
+ANSI_COLOR_WHITE=37
+ANSI_COLOR_BRIGHT_BLACK=90
+ANSI_COLOR_BRIGHT_RED=91
+ANSI_COLOR_BRIGHT_GREEN=92
+ANSI_COLOR_BRIGHT_YELLOW=93
+ANSI_COLOR_BRIGHT_BLUE=94
+ANSI_COLOR_BRIGHT_MAGENTA=95
+ANSI_COLOR_BRIGHT_CYAN=96
+ANSI_COLOR_BRIGHT_WHITE=97
+
+TRACE_LEVEL_QUIET=-1
+TRACE_LEVEL_STANDARD=0
+TRACE_LEVEL_VERBOSE=1
+TRACE_LEVEL_DEBUG=2
+TRACE_LEVEL=$TRACE_LEVEL_STANDARD
+
+function debug
+{
+	if [[ $TRACE_LEVEL -ge $TRACE_LEVEL_DEBUG ]]
+	then
+		echo -e "[\e[${ANSI_COLOR_GREEN}mDEBUG  \e[00m] $@"
+	fi
+}
+
+
+function info
+{
+	if [[ $TRACE_LEVEL -ge $TRACE_LEVEL_VERBOSE ]]
+	then
+		echo -e "[\e[${ANSI_COLOR_CYAN}mINFO   \e[00m] $@"
+	fi
+}
+
+function warning
+{
+	if [[ $TRACE_LEVEL -ge $TRACE_LEVEL_STANDARD ]]
+	then
+		echo -e "[\e[${ANSI_COLOR_YELLOW}mWARNING\e[00m] $@" 1>&2
+	fi
+}
+
+function error
+{
+	if [[ $TRACE_LEVEL -ge $TRACE_LEVEL_STANDARD ]]
+	then
+		echo -e "[\e[${ANSI_COLOR_BRIGHT_RED}mERROR  \e[00m] $@" 1>&2
+	fi
+}
+
+function fatal
+{
+	if [[ $TRACE_LEVEL -ge $TRACE_LEVEL_STANDARD ]]
+	then
+		echo -e "[\e[${ANSI_COLOR_RED}mFATAL  \e[00m] $@" 1>&2
+	fi
+	exit 1
+}
+
+function check_cmds
+{
+	for cmd in $@
+	do
+		{ hash $cmd > "/dev/null" 2>&1 ; } || { fatal "$cmd command not installed" ; }
+	done
+}
+
+function usage
+{
+	cat <<- EOF
+		usage: daos-launch.sh [OPTIONS] [-- ARGS]
+
+		Launch DAOS avocado functional tests
+
+		Options:
+		   -s, --start-cleanup      cleanup platform before starting tests
+		   -e, --end-cleanup        cleanup platform after running the tests
+		   -h, --help               show this help message and exit
+		   -V, --version            show version number
+		   -q, --quiet              quiet mode
+		   -v, --verbose            verbose mode
+		   -D, --debug              debug mode
+	EOF
+}
+
+function run
+{
+	if [[ $TRACE_LEVEL -ge $TRACE_LEVEL_STANDARD ]]
+	then
+		"$@"
+	else
+		"$@" &> /dev/null
+	fi
+}
+
+function cleanup
+{
+
+{% if "daos_clients" in groups and groups["daos_clients"] | length > 0 %}
+	info "Clean up of clients $CLIENTS_LIST"
+	$CLUSH_EXE -l root -w $CLIENTS_LIST sudo systemctl stop daos_agent
+	$CLUSH_EXE -l root -w $CLIENTS_LIST rm -rf /var/run/daos_agent
+	$CLUSH_EXE -l root -w $CLIENTS_LIST mkdir -p /var/run/daos_agent
+{% endif %}
+
+	info "Clean up of servers $SERVERS_LIST"
+	$CLUSH_EXE -l root -w $SERVERS_LIST sudo systemctl stop daos_server
+	$CLUSH_EXE -l root -w $SERVERS_LIST rm -rf /var/run/daos_server
+	$CLUSH_EXE -l root -w $SERVERS_LIST mkdir -p /var/run/daos_server
+
+	info "Stopping all DAOS processes"
+	{% if "daos_clients" in groups and groups["daos_clients"] | length > 0 %}
```

Comment:
(lint) You need a space after the '{'. [SC1054]
(lint) This } is literal. Check expression (missing ;/\n?) or quote it. [SC1083]